core/domain/app_feedback_report_domain.py

```diff
@@ -228,12 +257,14 @@ def require_valid_scrubber_id(cls, scrubber_id: str) -> None:
                 'The scrubbed_by user id %r is invalid.' % scrubber_id)
 
     @classmethod
-    def from_dict(cls, report_dict: Dict[str, Any]) -> AppFeedbackReport:
+    def from_dict(
```

Comment:
After changing the name from `from_dict()` to `from_submitted_feedback_dict()`, Schema validation fails because:

In `controllers/incoming_app_feedback_report.py` we are defining schema which says that incoming value has type `object_dict` https://github.com/oppia/oppia/blob/8e710ba0e6179e0cf5debc84c808a5cb40677a77/core/controllers/incoming_app_feedback_report.py#L35

And values of type `object_dict` are first initialized from their domain object using the `from_dict` method.
https://github.com/oppia/oppia/blob/8e710ba0e6179e0cf5debc84c808a5cb40677a77/core/schema_utils.py#L248

Since, after changing the name of `from_dict()` to `from_submitted_feedback_dict()`, the schema validator is unable to find the `from_dict` method to initialize the object and due to this, an error is showed up. 

Error messages that I get in the backend are: https://github.com/oppia/oppia/actions/runs/3039355879/jobs/4894192305#step:8:1179

Please correct me If I'm missing something.

Reply:
I think conceptually you can't validate the input as an AppFeedbackReport dict because it isn't one. So (a) this method name should change, and (b) the schema validation should be done in some other way (possibly manually).

Are there examples of schema validation where the controller just calls a custom validation function? Or, if the Android dict only has certain keys and values then specify those explicitly in the controller schema and don't use an object class.

/cc @Nik-09 who might also be able to help here.

Reply:
> /cc @Nik-09 who might also be able to help here.

An email was sent to Nikhil -- waiting for his reply, meanwhile can you please take a look at the other comments !!